package/MDAnalysis/coordinates/base.py

```diff
@@ -1014,6 +1021,63 @@ def add_auxiliary(self, auxname, auxdata, format=None, **kwargs):
         Auxiliary data is assumed to be time-ordered, with no duplicates. See
         the :ref:`Auxiliary API`.
         """
+        # auxdata can be an AuxReader or the data itself, try to get AuxReader
+        # from auxdata
+        try:
+            auxreader = get_auxreader_for(auxdata)
+        except AttributeError as e:
+            if str(e) == "'NoneType' object has no attribute 'upper'":
+                # user already gave instance of auxreader
+                auxreader = ""  # assign something so check below works
+
+        # Check if XVG format, call appropriate method
+        if isinstance(auxdata, _AUXREADERS["XVG"]) or \
```

Comment:
This is too complicated with all the isinstance checks, we need a more modular way to handle more readers. Possibly each reader could provide an attatchment method as a callback, then you would just have to do something like 

```
auxreader.attatch(self , ...)

```

This would require a consistent interface though.